### PR TITLE
[chatgpt]fix policy loss and value loss

### PR DESCRIPTION
**Pull request by wenjunyang**
## 📌 Checklist before creating the PR

- [x] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g.  #3078 

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

fix policy loss and value loss. 

## 💥 Checklist before requesting a review

- [x] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [ ] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: applications/ChatGPT/chatgpt/trainer/ppo.py [hunk 1]
```diff
@@ -75,25 +76,32 @@ def __init__(self,
...
-                                        experience.advantages,
+                                        advantage_with_penalty,
                                         action_mask=experience.action_mask)
         self.strategy.backward(actor_loss, self.actor, self.actor_optim)
         self.strategy.optimizer_step(self.actor_optim)
         self.actor_optim.zero_grad()
 
+        values = self.critic(experience.sequences,
+                             action_mask=experience.action_mask,
+                             attention_mask=experience.attention_mask)
```
Comment (ht-zhou) [inline applications/ChatGPT/chatgpt/trainer/ppo.py:100]: Why to calculate values for 2 times here?

Comment (wenjunyang) [reply applications/ChatGPT/chatgpt/trainer/ppo.py:100]: redundant, one should be removed

File: applications/ChatGPT/chatgpt/experience_maker/naive.py [hunk 2]
```diff
@@ -26,11 +26,9 @@ def make_experience(self, input_ids: torch.Tensor, **generate_kwargs) -> Experie
         value = self.critic(sequences, action_mask, attention_mask)
         r = self.reward_model(sequences, attention_mask)
 
-        reward = compute_reward(r, self.kl_coef, action_log_probs, base_action_log_probs, action_mask=action_mask)
```
Comment (wangcho2k) [inline applications/ChatGPT/chatgpt/experience_maker/naive.py:29]: Did you accidentally remove this?

Comment (wenjunyang) [reply applications/ChatGPT/chatgpt/experience_maker/naive.py:29]: no, My previous thought may be wrong. i will close the PR.

File: applications/ChatGPT/chatgpt/models/loss.py [hunk 3]
```diff
@@ -57,13 +57,9 @@ def __init__(self, clip_eps: float = 0.4) -> None:
 
     def forward(self,
                 values: torch.Tensor,
-                old_values: torch.Tensor,
                 reward: torch.Tensor,
                 action_mask: Optional[torch.Tensor] = None) -> torch.Tensor:
-        values_clipped = old_values + (values - old_values).clamp(-self.clip_eps, self.clip_eps)
```
Comment (ethanyanjiali) [inline applications/ChatGPT/chatgpt/models/loss.py:63]: @wenjunyang do you know why do we need to clip the value loss? and why are we taking the max of surr1 and surr2?

Comment (ethanyanjiali) [reply applications/ChatGPT/chatgpt/models/loss.py:63]: nvm got some answers here:
https://github.com/openai/baselines/blob/master/baselines/ppo2/model.py#L69-L75
https://github.com/openai/baselines/issues/91

Comment (wenjunyang) [reply applications/ChatGPT/chatgpt/models/loss.py:63]: thank you. it very helpful.
